react-app/src/components/projects/ProjectDetail.tsx

```diff
@@ -116,31 +124,56 @@ const ProjectDetail = (props: IProjectDetail) => {
           Name: 'Uncategorized',
         };
         if (!acc[fullStatus.Name]) {
-          acc[fullStatus.Name] = { Tasks: [{ ...curr, Name: fullTask.Name }], Notes: [] };
-          return acc;
-        } else {
-          acc[fullStatus.Name].Tasks.push({ ...curr, Name: fullTask.Name });
-          return acc;
+          acc[fullStatus.Name] = { Tasks: [], Notes: [], Timestamps: [], Monetaries: [] };
         }
+        acc[fullStatus.Name].Tasks.push({ ...curr, Name: fullTask.Name });
+        return acc;
       },
       {},
     );
-    return data?.parsedBody.Notes.filter((a) => a.Note).reduce(
+    data?.parsedBody.Notes.filter((a) => a.Note).reduce(
       (acc: Record<string, IStatusHistoryStruct>, curr) => {
         const fullNote = noteTypes.find((a) => a.Id === curr.NoteTypeId);
         const fullStatus = statuses.find((a) => a.Id === fullNote.StatusId) ?? {
           Name: 'Uncategorized',
         };
         if (!acc[fullStatus.Name]) {
-          acc[fullStatus.Name] = { Notes: [{ ...curr, Name: fullNote.Description }], Tasks: [] };
-          return acc;
-        } else {
-          acc[fullStatus.Name].Notes.push({ ...curr, Name: fullNote.Description });
-          return acc;
+          acc[fullStatus.Name] = { Tasks: [], Notes: [], Timestamps: [], Monetaries: [] };
         }
+        acc[fullStatus.Name].Notes.push({ ...curr, Name: fullNote.Description });
+        return acc;
```

Comment:
Is the acc here actually affecting the original reduceMap object? Tried finding documentation to see whether acc is a reference or a copy, but it's not clear.

Reply:
Yeah, it's affecting the original. Basically the first reduce starts with an empty object, puts a bunch of stuff in it, then returns a reference to that new object which we assign to reduceMap. Then we pass this reference as the starting accumulator to each following the reduce, and each one modifies it further. In JS, primitive types are passed by value, while object types (including arrays) are passed by reference.